src/connector/src/source/cdc/split.rs

```diff
@@ -124,27 +120,19 @@ impl PostgresCdcSplit {
         }
     }
 
-    pub fn copy_with_offset(&self, start_offset: String) -> Self {
-        // deserialize the start_offset
-        let dbz_offset: DebeziumOffset = serde_json::from_str(&start_offset)
-            .unwrap_or_else(|e| panic!("invalid cdc offset: {}, error: {}", start_offset, e));
-
-        let snapshot_done = dbz_offset
-            .source_offset
-            .last_snapshot_record
-            .unwrap_or(false);
-
-        let split = CdcSplitBase {
-            split_id: self.inner.split_id,
-            start_offset: Some(start_offset),
-            snapshot_done,
-        };
-
-        let server_addr = self.server_addr.clone();
-        Self {
-            inner: split,
-            server_addr,
+    pub fn update_with_offset(&mut self, start_offset: String) {
+        let mut snapshot_done = self.inner.snapshot_done;
+        if !snapshot_done {
+            let dbz_offset: DebeziumOffset = serde_json::from_str(&start_offset)
+                .unwrap_or_else(|e| panic!("invalid cdc offset: {}, error: {}", start_offset, e));
```

Comment:
a little -1 for panic here, shall we throw an error instead?

Reply:
fixed